### PR TITLE
[feature] use status-go signal subscriptions for current block

### LINKED ISSUE #8159: Use newBlockFilter signal from status-go
## Problem

Currently transactions confirmations is updated as part of the transaction fetching loop so it is delayed and based on information provided by a third party

## Rationale

While trivial, this issue will be done first as part of wallet improvement because it is the simplest use case for the newly added status-go subscriptions to eth filters

## Implementation

- Use status-go eth_newBlockFilter signal to update current block number
- base number of confirmations in transaction details on the current block number

## Acceptance Criteria

- number of confirmations in transaction details is updated within 2 seconds of a new block mined on chain

### PR DESCRIPTION
**Pull request by yenda**
fix #8159

- use signal subscriptions to get latest block signals
- compute confirmations in transaction details from current block
dynamically

# Testing

- tested on android emulator
- opened transaction list and checked transaction details for account `satoshi document engage inflict goddess auction rule unfair bid next buddy shy` on ropsten
- checked that number of confirmations is increased by one every time a new block is mined

status: ready

### CODE REVIEW

**Comment by yenda** [comment]
@mandrigin please notify me when new status-go is released

**Review by vitvly — APPROVED** [review]

**Review by cammellos — APPROVED** [review]

File: src/status_im/ethereum/subscriptions.cljs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  (:require [clojure.string :as string]
+            [re-frame.core :as re-frame]
+            [status-im.ethereum.decode :as decode]
+            [status-im.native-module.core :as status]
+            [status-im.utils.fx :as fx]
+            [taoensso.timbre :as log]))
+
+(defn get-block-by-hash [block-hash callback]
+  (status/call-private-rpc
+   (.stringify js/JSON (clj->js {:jsonrpc "2.0"
```
Comment (flexsurfer) [inline src/status_im/ethereum/subscriptions.cljs:11]: we have `types/clj->json` btw

File: src/status_im/ethereum/subscriptions.cljs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+                                 :id      1
+                                 :method  "eth_getBlockByHash"
+                                 :params  [block-hash false]}))
+   (fn [response]
+     (if (string/blank? response)
+       (log/warn :web3-response-error)
+       (callback (-> (.parse js/JSON response)
```
Comment (flexsurfer) [inline src/status_im/ethereum/subscriptions.cljs:18]: same here `json->clj`

**Review by flexsurfer — APPROVED** [review]

**Comment by churik** [comment]
Tested on IOS 11.4.1 (IPhone 7), Android 8 (LG V20):
- transaction view is updated for outgoing and incoming transactions in STT and EThro
- can send transaction by scanning QR, from test Dapp
- wallet balance is updated

**Pull request merged**